### PR TITLE
Snowman_Escape

### PR DESCRIPTION
**Pull request by DefinitelyANon**
# Your checklist for this pull request

## Author name

**Author: Michael Q**

## About your game

**What is your game about?**

You are a snowman, and you must reach the goal of each level without dying.

**How do you play your game?**

Use WASD to move, J to restart.
Reach the goal (green circle) on each level. Rocks are can be pushed. if there are any buttons (blue circles), a rock must be on top of it for the goal to work. Don't walk into lasers.

## Code
Check off the items that are true.
- [x] The game was made using the [Sprig editor](https://sprig.hackclub.com/editor).
- [x] The game is placed in the in the [`/games` directory](https://github.com/hackclub/sprig/tree/main/games).
- [x] The code is significantly different from all other games in the [Sprig gallery](https://sprig.hackclub.com/gallery) (except for games labeled "demo").
- [x] The game runs without errors. 
- [x] The name of the file/game contains only alphanumeric characters, `-`s, or `_`s.
- [x] The game name is not the same as the others from [gallery](https://sprig.hackclub.com/gallery)

> Thanks for your PR!

### CODE REVIEW

**Review by LucasHT22 — CHANGES_REQUESTED** [review]
Thanks for PR! Where did you hear about Sprig?

**Comment by DefinitelyANon** [comment]
> Thanks for PR! Where did you hear about Sprig?

I heard about it from a friend, who also made a game on Sprig.

**Comment by LucasHT22** [comment]
> > Thanks for PR! Where did you hear about Sprig?
> 
> I heard about it from a friend, who also made a game on Sprig.

oh nice! What is their game? Sorry for the delay too

**Comment by DefinitelyANon** [comment]
> > > Thanks for PR! Where did you hear about Sprig?
> > 
> > 
> > I heard about it from a friend, who also made a game on Sprig.
> 
> oh nice! What is their game? Sorry for the delay too

No worries! Their game is called "starjump", made by skifli.

File: games/Snowman_Escape.js [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+      const tile = getTile(x, y)
+      const playerb = getFirst(player)
+      const goalb = getFirst(goal)
+
+      if (tile.length == 0) {
+        addSprite(x, y, laser)
+      }
+        
+      /*There is definitely a better way to do this, */
+      /*but all things I tried didn't work because JS scares me*/
```
Comment (LucasHT22) [inline games/Snowman_Escape.js:326]: Haha but JS isnt a snake like others ;)

**Review by LucasHT22 — CHANGES_REQUESTED** [review]
Out of curiosity, did you use any game for inspiration?

File: games/Snowman_Escape.js (hunk 1 above)
Comment (DefinitelyANon) [reply games/Snowman_Escape.js:326]: Haha lol, at least the Python looks pretty cool. Still wouldn’t want to come across one in person, though.

**Comment by DefinitelyANon** [comment]
> Out of curiosity, did you use any game for inspiration?

Not any in particular, I just thought it would be cool to add lasers to the template sokoban game.
Although I didn’t change things up in my game too much, I really admired how “Baba is me” played with the traditional sokoban formula to make their game.

**Review by LucasHT22 — CHANGES_REQUESTED** [review]
Right:) How to solve this level? 
image

**Comment by DefinitelyANon** [comment]
> Right:) How to solve this level? image

Ah, this was a tough one! :)
(Just in case - the blue rock has to be pushed onto the blue circle before the green circle (aka the goal) will activate.)

**Comment by LucasHT22** [comment]
Thanks! And this?
image

**Comment by DefinitelyANon** [comment]
> Thanks! And this? image

Yeah, this level is quite brutal. I've (hopefully) uploaded a video here that shows the completion of that level and the last few remaining. Hope this helps!

https://github.com/hackclub/sprig/assets/136814442/05cbd805-8dfa-43fa-b9d9-daf8ebfa9c3d

**Review by LucasHT22 — APPROVED** [review]
Congrats:)
If you're a teen, you can get your sprig here! https://sprig-order.hackclub.dev/

Hack Clubbers would love to see this! Just post a link and 1-2 sentences talking about your game to our [#ship](https://hackclub.slack.com/archives/C0M8PUPU6) [channel](https://hackclub.com/slack/):)

**Pull request merged**